### PR TITLE
Added bstd_subtract for bstd_number

### PR DESCRIPTION
**Pull request by MartvanAssen**

### CODE REVIEW

File: src/arithmetic.c [hunk 1]
```diff
@@ -28,6 +28,20 @@ void bstd_add(bstd_number *lhs, const bstd_number *rhs) {
     free((void*)sum);
 }
 
+void bstd_subtract(bstd_number *lhs, const bstd_number *rhs) {
+    bstd_number rhs_copy = *rhs;
```
Comment (EppeMedia) [inline src/arithmetic.c:32]: This is not how copying works, we are violating the "const" promise

File: src/arithmetic.c [hunk 2]
```diff
@@ -28,6 +28,20 @@ void bstd_add(bstd_number *lhs, const bstd_number *rhs) {
...
+    rhs_copy.isSigned = true;
+    if (rhs_copy.positive == 0) {
+        rhs_copy.positive = 1;
+    } else {
+        rhs_copy.positive = 0;
+    }
```
Comment (EppeMedia) [inline src/arithmetic.c:38]: The if/else could be be a single negation

File: include/arithmetic.h [hunk 3]
```diff
@@ -18,6 +18,14 @@ int64_t ipow(int64_t base, uint64_t exp);
  */
 void bstd_add(bstd_number *lhs, const bstd_number *rhs);
 
+/**
+ * subtracts the specified left- and right-hand sides, and assigns the result to the specified left-hand side.
+ * This function does not modify the right-hand side.
```
Comment (EppeMedia) [inline include/arithmetic.h:23]: See comment below :)

**Review by EppeMedia — CHANGES_REQUESTED** [review]

File: src/arithmetic.c (hunk 1 above)
Comment (MartvanAssen) [reply src/arithmetic.c:32]: This is a valid way to copy as long as the struct we copy contains no pointers.

https://stackoverflow.com/questions/9127246/copy-struct-to-struct-in-c

Comment (EppeMedia) [reply src/arithmetic.c:32]: You're absolutely right, number does not contain pointers

File: include/arithmetic.h (hunk 3 above)
Comment (EppeMedia) [reply include/arithmetic.h:23]: That comment is resolved.

**Comment by EppeMedia** [comment]
Duplicate work. The other one is merged, this can be closed.

**Pull request closed without merging**